proto/src/determined/resourcepool/v1/resourcepool.proto

```diff
@@ -0,0 +1,182 @@
+syntax = "proto3";
+
+package determined.resourcepool.v1;
+option go_package = "github.com/determined-ai/determined/proto/pkg/resourcepoolv1";
+
+
+
+// A Resource Pool is a pool of resources where containers are run.
+message ResourcePool {
+  // The unique name of the resource pool.
+  string name = 1;
+  // The description of the resource pool
+  string description = 2;
+  // The type of the resource pool (AWS/GCP/static)
+  string type = 3;
```

Comment:
question: could this be defined as an enum?

Reply:
same for scheduler type, scheduler fitting policy,

Reply:
Good idea. Changed these to enums and the code is much better.

Reply:
Hmm, protobuf enums + buf lint actually kind of sucks. I'm using enums, but I really don't like what it looks like

Reply:
> protobuf enums + buf lint actually kind of sucks

what did you run into?